test/integration/targets/vmware_tag/tasks/tag_crud_ops.yml

```diff
@@ -0,0 +1,94 @@
+# Test code for the vmware_tag CRUD Operations.
+# Copyright: (c) 2019, Pavan Bidkar <pbidkar@vmware.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Wait for Flask controller to come up online
+  wait_for:
+    host: "{{ vcsim }}"
+    port: 5000
+    state: started
+
+- name: kill vcsim
+  uri:
+    url: http://{{ vcsim }}:5000/killall
+
+- name: start vcsim
+  uri:
+    url: http://{{ vcsim }}:5000/spawn?cluster=2
+  register: vcsim_instance
+
+- debug: var=vcsim_instance
+
+- name: Wait for vcsim server to come up online
+  wait_for:
+    host: "{{ vcsim }}"
+    port: 443
+    state: started
+
+- name: Create Category
+  vmware_category:
+    hostname: "{{ vcsim }}"
+    username: "{{ vcsim_instance['json']['username'] }}"
+    password: "{{ vcsim_instance['json']['password'] }}"
+    validate_certs: False
+    category_name: Sample_Cat_0006
+    category_description: Sample Description
+    category_cardinality: 'multiple'
+    state: present
+  register: category_create
+
+- name: Check Category is created
+  assert:
+    that:
+      - "{{ category_create.changed == true }}"
+
+- name: Set Cat_ID Paramter. Required for Tag creation
+  set_fact: Cat_ID={{ category_create['category_results']['category_id'] }}
+
+- name: Create a tag
```

Comment:
Run this task again to check idempontency of vmware_tag

Reply:
Tested as standalone playbook in local test env